### PR TITLE
[MINOR] Fix Typos 'a -> an'

### PR DESCRIPTION
**Pull request by zhengruifeng**
## What changes were proposed in this pull request?

`a` -> `an`

I use regex to generate potential error lines:
`grep -in ' a [aeiou]' mllib/src/main/scala/org/apache/spark/ml/*/*scala`
and review them line by line.
## How was this patch tested?

local build
`lint-java` checking

### CODE REVIEW

**Comment by holdenk** [comment]
So it seems that in a few places adding the extra character has pushed it over the 100. You should probably run the linter explicitly if you have it disabled by default `./dev/lint-scala`.

File: core/src/main/scala/org/apache/spark/MapOutputTracker.scala [hunk 1]
```diff
@@ -105,7 +105,7 @@ private[spark] abstract class MapOutputTracker(conf: SparkConf) extends Logging
     }
   }
 
-  /** Send a one-way message to the trackerEndpoint, to which we expect it to reply with true. */
+  /** Send an one-way message to the trackerEndpoint, to which we expect it to reply with true. */
```
Comment (holdenk) [inline core/src/main/scala/org/apache/spark/MapOutputTracker.scala]: I don't think this change is correct.

**Comment by zhengruifeng** [comment]
@holdenk Thanks. I have fixed this. and run `lint-java` to check java file.

**Comment by holdenk** [comment]
Also, your change seems to have made a few odd changes "an one way" which sounds odd, generally "a one way" is considered sounding "better" (I'm a bit fuzzy on the exact rule - but if you look you'll see people say "a one way ticket" instead of "an one way ticket" and some other similar things).

**Comment by zhengruifeng** [comment]
@holdenk Thanks. I think you are right. I will revert `an one-xxx` to `a one-xxx`.

**Comment by holdenk** [comment]
Also if this was created by a regex or similar it might be useful for review to also mention the rule that was applied to make the change.

**Comment by zhengruifeng** [comment]
@holdenk Yes. I use cmd like this `grep -in ' a [aeiou]' mllib/src/main/scala/org/apache/spark/ml/*/*scala` 
to generate potential wrong lines.

**Comment by zhengruifeng** [comment]
@holdenk Excuse me,  How to deal with MiMa test failures? It seems caused by change in `core`

**Comment by holdenk** [comment]
Oh I think the MiMa test failures are spurious. Personally I'm going to go to sleep and trigger a test again in the morning.

**Comment by zhengruifeng** [comment]
@holdenk Thanks. Good night.

**Comment by srowen** [comment]
As a grammar pedant, I approve, even though this is does end up touching a lot of files. It's helpful to just say in the title that the change is "a -> an" fixes but no big deal.

**Comment by holdenk** [comment]
So when I was skimming it last night, it looked like it also drops "a and" to "and" (which is good) but we might also want to mention.

**Comment by rxin** [comment]
Merging in master/2.0.

**Pull request closed without merging**